src/components/hooks/use-browser-resize.js

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from 'react'
+import { isBrowser } from 'common/utility'
+import { size } from 'themes/device'
+
+export const useBrowserResize = (maxSize) => {
+    const [is_mobile, setMobile] = useState(false)
+    const [width, setWidth] = useState(isBrowser() ? window.innerWidth : 0)
+    const breakpoint = maxSize ? maxSize : size.tablet
+
+    useEffect(() => {
+        if (!isBrowser()) return false
+        const handleResize = () => setWidth(window.innerWidth)
+        handleResizeWindow()
+        window.addEventListener('resize', handleResize)
+
+        return () => {
+            window.removeEventListener('resize', handleResize)
+            handleResizeWindow()
+        }
+    })
```

Comment:
```suggestion
    useEffect(() => {
        if (!isBrowser()) return false
        const handleResize = () => setWidth(window.innerWidth)
        handleResizeWindow()
        window.addEventListener('resize', handleResize)

        return () => {
            window.removeEventListener('resize', handleResize)
            handleResizeWindow()
        }
    }, [])
```

Reply:
We only want to add the event listener on mount right?

Reply:
We want to use this hook for everytime user resize their browser and not just once..